### ISSUE TITLE
Issue after upgrading to 0.7.2 from 0.7.0

### ISSUE THREAD

**Issue by jochenjonc** [JasonWeinzierl/eslint-plugin-rxjs-x#193]
Hi,

The change in #185 gives another kind of false positive, which I think worked before.

When you have a function with multiple signatures, like this:

```typescript
export interface Hook {
  onInit?: (field: Field) => void | ((field: Field) => Observable<any>);
}
```

It doesn't seem to be able to distinguish between the one that returns void and the one that returns the Observable.

When I reorder it, like this:

```typescript
export interface Hook {
  onInit?: ((field: Field) => Observable<any>) | (field: Field) => void;
}
```

It does work (I don't have a linting error).

But the sample I give is not part of code that I have under control, it comes from the firmly library. See https://github.com/ngx-formly/ngx-formly/blob/main/src/core/src/lib/models/fieldconfig.ts#L249-L257

Would it be possible to handle such cases in eslint-plugin-rxjs-x?

**Comment by JasonWeinzierl** [comment]
Is this for the `finnish` rule?

**Comment by JasonWeinzierl** [comment]
(Might be unrelated but documenting this here: seems there's no `finnish` test coverage for arrow function declarations. VariableDeclarator > Identifier to couldReturnObservable is always false.  Not sure if this is missing test coverage or intentional behavior?)

**Comment by jochenjonc** [comment]
It's not for the `finnish` rule, but for the `no-misused-observables`.

**Comment by jochenjonc** [comment]
Hi,

The scenario you are describing works, but this little twist doesn't:

```typescript
      import { Observable, of } from "rxjs";

      export type FormlyHookFn = (field: FormlyFieldConfig) => void;

      interface Hook {
        onInit?: FormlyHookFn | ((field: string) => Observable<any>);
      }

      const hook: Hook = {
        onInit: field => of(field),
      };
```

Regards,

Jochen
